notification-server/src/google_cloud.rs

```diff
@@ -315,6 +316,58 @@ mod tests {
         }
     }
 
+    #[tokio::test]
+    async fn test_send_push_notification() {
+        let mut server = mockito::Server::new_async().await;
+        let mock_provider = MockTokenProvider {
+            token_response: Arc::new("mock_token".to_string()),
+            should_fail:    false,
+        };
+
+        let expected_body = json!({
+            "message": {
+                "token": "valid_device_token",
+                "data": {
+                    "recipient": "4FmiTW2L2AccyR9VjzsnpWFSAcohXWf7Vf797i36y526mqiEcp",
+                    "amount": "100",
+                    "type": "ccd-tx"
+                }
+            }
+        });
+
+        let mock = server
+            .mock("POST", "/v1/projects/fake_project_id/messages:send")
+            .match_body(mockito::Matcher::Json(expected_body))
+            .with_status(200)
+            .with_body(json!({"success": true}).to_string())
+            .expect(1)
+            .create_async()
+            .await;
+
+        let client = Client::new();
+        let backoff_policy = ExponentialBackoff {
+            max_elapsed_time: Some(Duration::from_millis(50)),
+            max_interval: Duration::from_millis(1),
+            initial_interval: Duration::from_millis(1),
+            ..ExponentialBackoff::default()
+        };
+        let mut gc = GoogleCloud::new(client, backoff_policy, mock_provider, "mock_project_id");
+        gc.url = format!(
+            "{}{}",
+            server.url(),
+            "/v1/projects/fake_project_id/messages:send".to_string()
+        );
+        let notification_information = NotificationInformation::new(
+            AccountAddress::from_str("4FmiTW2L2AccyR9VjzsnpWFSAcohXWf7Vf797i36y526mqiEcp").unwrap(),
+            "100".to_string(),
+            Preference::CCDTransaction,
+        );
+        assert!(gc
+            .send_push_notification("valid_device_token", notification_information)
+            .await
+            .is_ok());
+        mock.assert();
+    }
```

Comment:
What are we even testing here? To me it seems like you create a mock server which will always return a success response only to test that it returns a success? Or am I missing something?

Reply:
This here is indeed a super important test. This is the contract which ensures we align with what the client expect. 

In particular this line:

`.match_body(mockito::Matcher::Json(expected_body))`